certora/specs/RevertsMintBurnOptimism.spec

```
@@ -8,24 +8,17 @@ methods {
     function _._moveDelegateVotes(address, address, uint256) internal => CONSTANT;
 }
 
-// Check that minting to zero address or msg.sender is not bridge revert.
-rule mintReverts(env e, address to, uint256 amount) {
-    // Safe require as mint is a non-payable function.
-    require e.msg.value == 0;
-
-    require totalSupply() + amount <= max_uint256;
+// Check the revert conditions for the mint function.
+rule mintRevertConditions(env e, address to, uint256 amount) {
+    mathint totalSupplyBefore = totalSupply();
 
     mint@withrevert(e, to, amount);
-    assert   !(lastReverted) <=> e.msg.sender == currentContract.bridge && to != 0;
+    assert lastReverted <=> e.msg.sender != currentContract.bridge || to == 0 || e.msg.value != 0 || (totalSupplyBefore + amount) > max_uint256;
```

```suggestion
    assert lastReverted <=> e.msg.sender != currentContract.bridge || to == 0 || e.msg.value != 0 || totalSupplyBefore + amount > max_uint256;
```
to be consistent with what is done in the mint rule for the Ethereum token

---

Fixed in https://github.com/morpho-org/morpho-token-upgradeable/pull/98/commits/9025f3b5c34e16082305e6d4b97dc5bf96eb67f5.